src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSpark.java

```diff
@@ -23,14 +23,23 @@
 
     private static final long serialVersionUID = 1L;
 
+    public static final String SINGLE_END_ALIGNMENT_FULL_NAME = "singleEndAlignment";
+    public static final String SINGLE_END_ALIGNMENT_SHORT_NAME = "SE";
+
     @Argument(doc = "the output bam", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
             fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
     private String output;
 
     @Argument(doc = "the bwa mem index image file name that you've distributed to each executor",
-            fullName = "bwamemIndexImage")
+            fullName = "bwaMemIndexImage")
     private String indexImageFile;
```

Comment:
Now that the indexing code is available in the JNI, I think that it will be nice to add a mutex argument to this one to provide the reference sequence instead: if this argument is provided, the pre-generated image will be used; otherwise, it will create the image file from the reference sequence.

Like that, the BwaSpark tool can be self-contained and be used directly with input reads and a reference sequence, without any additional parameter.